modules/ballerina-core/src/main/java/org/ballerinalang/model/values/BArray.java

```diff
@@ -51,24 +53,34 @@ public BArray(Class<V> valueClass) {
         this.valueClass = valueClass;
     }
 
-    public <V extends BValue> void add(int index, V value) {
-        ensureCapacity(index);
+    public <V extends BValue> void add(long index, V value) {
+        if (index != (int) index) {
```

Comment:
Actually, java 8 has Math.toIntExact(long value) method to do this, it uses the same logic used here, And if we are to check index > Integer.MAX_VALUE, then we will have to do it for both sides
index < Integer.MIN_VALUE, and it is also possible.